File: note-keeper-service/src/main/java/xyz/crearts/note/keeper/dto/TodoInput.java
package xyz.crearts.note.keeper.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class TodoInput {
    @NotBlank(message = "Title is required")
    private String title;
    private String description;
    private Boolean completed;
    private List<String> tags;
    private String priority;
    private Boolean isFavorite;
    
    private String dueDate;
    private String reminder;
    private String notificationChannels;
    
    private Map<String, Object> location;
    private Map<String, Object> schedule;
    private List<Map<String, Object>> attachments;
}

File: android/app/src/main/java/xyz/crearts/notekeeper/data/model/TodoInput.kt
package xyz.crearts.notekeeper.data.model

data class TodoInput(
    val title: String,
    val description: String? = null,
    val tags: List<String>? = null,
    val priority: String? = null,
    val isFavorite: Boolean? = null,
    val completed: Boolean? = null,
    val dueDate: String? = null,
    val reminder: String? = null,
    val notificationChannels: String? = null,
    val location: LocationInput? = null,
    val schedule: ScheduleInput? = null,
    val attachments: List<AttachmentInput>? = null
)

data class LocationInput(
    val lat: Double? = null,
    val lng: Double? = null,
    val address: String? = null
)

data class ScheduleInput(
    val repeat: String,
    val endDate: String? = null
)
